compiler/passes/scopeResolve.cpp

```diff
@@ -2042,6 +2114,73 @@ static void buildBreadthFirstModuleList(Vec<ModuleSymbol*>* modules,
   }
 }
 
+// Returns true if this module is capable of being used or traversed as part of
+// an access in the provided scope, false if the module is private and the
+// scope is not its direct parent or a sibling module when the parent of both
+// had a use of it.
+bool ModuleSymbol::isVisible(BaseAST* scope) {
+  if (hasFlag(FLAG_PRIVATE)) {
+    BaseAST* parentScope = getScope(defPoint);
+    INT_ASSERT(parentScope != NULL); // Should be true, given we found this
+    // module symbol.
+
```

Comment:
I may well be confused but I wonder if it would be better to eliminate the special test on line 2127 and then
initialize searchScope to be scope in line 2136?  I think the functionality would be the same (?) and would be
surprised if there were an observable performance difference for any non-byzantine test.  I think flattening the
if/else would make the business logic slightly more consistent/evident.